File: results/RQ1/05_baseline_qualification/units/p04_caro_game/python/replays/0001_seed_812711061/repository/caro/move.py
"""Immutable record for tracking moves in the Caro game."""

from dataclasses import dataclass
from .types import Player


@dataclass(frozen=True)
class Move:
    """Represents a move made by a player at a specific location on the board.

    This is an immutable record that captures the essential information about
    a move for history tracking purposes.
    """
    row: int
    col: int
    player: Player

File: results/RQ1/05_baseline_qualification/units/p04_caro_game/cpp/attempts/0004_seed_3838723605/p04_caro_game_cpp/p04_caro_game_cpp/src/Move.cpp
#include "Move.h"

namespace FiveInARow {

Move::Move(size_t r, size_t c, Player p) : row(r), col(c), player(p) {}

bool Move::operator==(const Move& other) const noexcept {
    return row == other.row && col == other.col && player == other.player;
}

}  // namespace FiveInARow